pkg/cli/cli.go

```diff
@@ -193,9 +255,13 @@ func (c cli) validate() error {
 	}
 
 	// Validate plugin versions and name.
-	for _, versionedPlugins := range c.plugins {
+	for _, versionedPlugins := range c.pluginsFromOptions {
+		pluginNameSet := map[string]struct{}{}
 		for _, versionedPlugin := range versionedPlugins {
 			pluginName := versionedPlugin.Name()
+			if err := plugin.ValidateName(pluginName); err != nil {
+				return fmt.Errorf("failed to validate plugin name %q: %v", pluginName, err)
```

Comment:
This code shows duplicated. 

Could we not:
- Get the plugin that will be built stored and save (new one)
- Then, get all project plugins and check if has the same name and version

Reply:
Duplicated? How do you mean?

Reply:
The same checks in the inside of both the loops are identical. See that the lines https://github.com/kubernetes-sigs/kubebuilder/blob/2003c53102f972a8b3b65ca3cb38014dd3dc6d36/pkg/cli/cli.go#L261-L269 are the same as https://github.com/kubernetes-sigs/kubebuilder/blob/2003c53102f972a8b3b65ca3cb38014dd3dc6d36/pkg/cli/cli.go#L287-L296 which lead us that could be an implementation centralized and encapsulated. 

Why do we need to re-check the name, version and etc of all plugins that are stored in the Project file?
IHMO, we need to check the current input data only. I mean just the command/plugin which was called and if it is or not supported for the project version.

Reply:
These are checking two different `cli` fields: `cliPluginKeys` and `pluginsFromOptions`. They have different types (`string` vs `plugin.Base`); combining keys from `pluginsFromOptions` would still require a loop, and we'd lose error information. I'd like to keep field validation separate as much as possible, if not for readability then for proper error reporting.